google/resource-snippets/pubsub-v1/pubsub.jinja

```diff
@@ -12,21 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{% set topic = 'configurable-pubsub-topic-' + env['deployment'] %}
-{% set subscription = env['deployment'] + '-' + env['name'] + '-subscription' %}
-
 resources:
-- name: my-topic
-  #type: pubsub.v1.topic
-  type: gcp-types/pubsub-v1:projects.topics
-  properties:
-    topic: {{ topic }}
-- name: my-subscription
-  #type: pubsub.v1.subscription
-  type: gcp-types/pubsub-v1:projects.subscriptions
-  properties:
-    subscription: {{ subscription }}
-    topic: $(ref.my-topic.name)
-    {% if "ackDeadlineSeconds" in properties %}
-    ackDeadlineSeconds: {{ properties["ackDeadlineSeconds"] }}
-    {% endif %}
+  - type: pubsub.v1.topic
+    name: {{ properties["topic_name"] }}
+    properties:
+      topic: my-pubsub-topic
+  - type: pubsub.v1.topic
+    name: {{ properties["backup_topic_name"] }}
+    properties:
+      topic: my-backup-topic
+  - type: pubsub.v1.subscription
+    name: {{ properties["subscription_name"] }}
+    properties:
+      subscription: my-pubsub-subscription
+      topic: $(ref.{{properties["topic_name"]}}.name)
+      {% if "ackDeadlineSeconds" in properties %}
+      ackDeadlineSeconds: {{ properties["ackDeadlineSeconds"] }}
+      {% endif %}
+
+      {% if "messageRetentionDuration" in properties %}
+      messageRetentionDuration: {{ properties["messageRetentionDuration"] }}
+      {% endif %}
+      {% if "retainAckedMessages" in properties %}
+      retainAckedMessages: {{ properties["retainAckedMessages"] }}
+      {% endif %}
+
+      {% if "deadLetterPolicy" in properties %}
+      deadLetterPolicy:
+       deadLetterTopic: $(ref.{{properties["backup_topic_name"]}}.name)
+       maxDeliveryAttempts: {{ properties["deadLetterPolicy"].maxDeliveryAttempts }}
+      {% endif %}
+
+      {% if "retryPolicy" in properties %}
+      retryPolicy:
+       minimumBackoff: {{ properties["retryPolicy"].minimumBackoff }}
+       maximumBackoff: {{ properties["retryPolicy"].maximumBackoff}}
```

Comment:
Nit: This file could use some formatting using a tool i.e. `maximumBackoff}}` -> `maximumBackoff }}` and a few other places in this file.

The indenting here is also only one space, and our standard is 2 spaces